### PR TITLE
Added additional doctrine event to see if fileset is done, THEN zip f…

### PR DESCRIPTION
**Pull request by mickel1138**
…iles.

### CODE REVIEW

File: src/MessageHandler/ZipDatasetFilesHandler.php
Comment (praneethpr) [inline src/MessageHandler/ZipDatasetFilesHandler.php:86]: Annotation missing for $entityEventDispatcher

File: src/MessageHandler/ZipDatasetFilesHandler.php
Comment (praneethpr) [inline src/MessageHandler/ZipDatasetFilesHandler.php:122]: I think we can remove this, and add it when we are going to listen for this event?

Comment (mickel1138) [reply src/MessageHandler/ZipDatasetFilesHandler.php:122]: Not every event always has a listener. But in our case we might remove it.

**Review by praneethpr — APPROVED** [review]

File: src/Event/FileProcessListener.php
Comment (fightingtexasaggie) [inline src/Event/FileProcessListener.php:75]: Since this is getting called with every single file, this method is going to get called a lot, I worry a little about the efficiency of the criteria call, but I guess we can optimize it later if it becomes an issue.

**Review by fightingtexasaggie — APPROVED** [review]
I'm assuming y'all decided not to wait for dataset review to be accepted before generation of the zipfile, because that's how it's currently working. Approving contingent on that being the intent.

File: src/Event/FileProcessListener.php
Comment (mickel1138) [reply src/Event/FileProcessListener.php:75]: This call is being resolved in memory, 

> I'm assuming y'all decided not to wait for dataset review to be accepted before generation of the zipfile, because that's how it's currently working. Approving contingent on that being the intent.

No, approval is not a factor in zipping.

**Comment by fightingtexasaggie** [comment]
I just re-tested with 1GB of data spread across 100 files and quite a few of them were skipped. I'm thinking maybe there was an http timeout, but will now retest in async mode.

**Pull request merged**
